core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6981ProjectListShouldIncludeChildrenTest.java

```diff
@@ -0,0 +1,68 @@
+package org.apache.maven.it;
+
+import org.apache.maven.it.util.ResourceExtractor;
+
+import java.io.File;
+import java.util.List;
+
+public class MavenITmng6981ProjectListShouldIncludeChildrenTest
+        extends AbstractMavenIntegrationTestCase
+{
+
+    private static final String RESOURCE_PATH = "/mng-6981-pl-should-include-children";
+
+    public MavenITmng6981ProjectListShouldIncludeChildrenTest()
+    {
+        super( "[3.7.0,)" );
+    }
+
+    public void testProjectListShouldIncludeChildrenByDefault()
+            throws Exception
+    {
+        final File testDir = ResourceExtractor.simpleExtractResources( getClass(), RESOURCE_PATH );
+        Verifier verifier = newVerifier( testDir.getAbsolutePath() );
+
+        verifier.addCliOption( "-pl" );
+        verifier.addCliOption( ":module-a" );
+        verifier.executeGoal( "compile" );
+        verifier.verifyTextInLog( "Building module-a-1 1.0" );
+    }
+
+    /**
+     * Since --pl's behavior is changed, make sure the alternative for building a pom without it's children still works.
```

Comment:
I think this should be in the documentation somewhere, too. Not just here in a Javadoc, but maybe in the [Guide to Working with Multiple Modules](https://maven.apache.org/guides/mini/guide-multiple-modules.html)?

Reply:
Good idea. I have created a [PR on maven-site](https://github.com/apache/maven-site/pull/199) about this alternative, but also some other docs that were missing. Is this what you had in mind?

I think we should extend the PR with more documentation related to the other reactor changes coming in 3.7.0.